testgen/generators.go

```diff
@@ -585,30 +796,29 @@ var EthGetBlockTransactionCountByHash = MethodTests{
 			"get-genesis",
 			"gets tx count in block 0",
 			func(ctx context.Context, t *T) error {
-				block := t.chain.GetBlockByNumber(0)
+				block := t.chain.GetBlock(0)
 				var got hexutil.Uint
 				err := t.rpc.CallContext(ctx, &got, "eth_getBlockTransactionCountByHash", block.Hash())
 				if err != nil {
 					return err
 				}
-				want := len(t.chain.GetBlockByNumber(0).Transactions())
-				if int(got) != want {
-					return fmt.Errorf("tx counts don't match (got: %d, want: %d)", int(got), want)
+				if int(got) != 0 {
+					return fmt.Errorf("tx counts don't match (got: %d, want: %d)", int(got), 0)
 				}
 				return nil
 			},
 		},
 		{
 			"get-block-n",
-			"gets tx count in block 2",
+			"gets tx count in a non-empty block",
 			func(ctx context.Context, t *T) error {
-				block := t.chain.GetBlockByNumber(2)
+				block := t.chain.BlockWithTransactions("", nil)
```

Comment:
```suggestion
				block := t.chain.BlockWithTransactions("any", nil)
```
maybe this for clarity?